Source/Core/Core/Slippi/SlippiMatchmaking.cpp

```diff
@@ -312,15 +323,23 @@ void SlippiMatchmaking::startMatchmaking()
   else
   {
     host_entry = gethostbyname(host);  // find host information
-    if (host_entry == NULL)
+    if (host_entry == NULL || host_entry->h_addrtype != AF_INET)
     {
       ERROR_LOG(SLIPPI_ONLINE, "[Matchmaking] Error finding LAN host");
     }
     else
     {
-      ip = inet_ntoa(*((struct in_addr*)host_entry->h_addr_list[0]));  // Convert into IP string
-      INFO_LOG(SLIPPI_ONLINE, "[Matchmaking] LAN IP: %s", ip);
+      // Fetch the last IP (because that was correct for me, not sure if it will be for all)
+      int i = 0;
+      while (host_entry->h_addr_list[i] != 0)
+      {
+        ip = inet_ntoa(*((struct in_addr*)host_entry->h_addr_list[i]));
```

Comment:
pretty sure this is deprecated by `inet_ntop`

Reply:
some documentation:
https://docs.microsoft.com/en-us/windows/win32/api/ws2tcpip/nf-ws2tcpip-inet_ntop
https://www.qnx.com/developers/docs/6.5.0SP1.update/com.qnx.doc.neutrino_lib_ref/i/inet_ntop.html

prefer using the idiomatic pattern of checking return value of function != null for success, and passing in destination buffer in the params instead of doing assignment